spec/features_spec.rb

```diff
@@ -1,11 +1,42 @@
 # (c) 2017 Ribose Inc.
 #
 
+# No point in using ApplicationRecord here.
+# rubocop:disable Rails/ApplicationRecord
+
 require "spec_helper"
 
 RSpec.describe "Attr Masker gem" do
-  example "The Rake task doesn't raise any exceptions" do
-    expect { run_rake_task }.not_to raise_exception
+  before do
+    stub_const "User", Class.new(ActiveRecord::Base)
+  end
+
+  let!(:han) do
+    User.create!(
+      first_name: "Han",
+      last_name: "Solo",
+      email: "han@example.test",
+    )
+  end
+
+  let!(:luke) do
+    User.create!(
+      first_name: "Luke",
+      last_name: "Skywalker",
+      email: "luke@jedi.example.test",
+    )
+  end
+
+  example "Masking a single text attribute with default options" do
+    User.class_eval do
+      attr_masker :last_name
+    end
+
+    expect { run_rake_task }.not_to(change { User.count })
+
+    expect { han.reload }.to(change { han.last_name }.to("(redacted)"))
```

Comment:
@abunashir I've started with feature specs because we need some end-to-end tests, and they will be the most useful type of tests during upcoming refactoring which will alter everything and will be pretty close to rewrite. I am going to add unit tests for particular classes when rewriting them.